Resources/CoreData/AtomicModules/AtomicEventLoop.js

```diff
@@ -0,0 +1,475 @@
+        cb_func = func;
+    }
+
+    timer_id = evloop.nextTimerId++;
+
+    evloop.insertImmediate({
+        id: timer_id,
+        oneshot: true,
+        cb: cb_func
+    });
+
+    return timer_id;
+}
+
+/**
+ * stops a previously queued setImmediate callback.
+ * @method
+ * @param {int} timer_id the id of the timer that was created via setImmediate
+ */
+function clearImmediate(timer_id) {
+    var evloop = EventLoop;
+
+    if (typeof timer_id !== 'number') {
+        throw new TypeError('timer ID is not a number');
+    }
+    evloop.removeTimerById(timer_id, true);
+}
+
+/**
+ * set up a timer that repeats every "delay" milliseconds.
+ * @method
+ * @param {function} func callback function to call every "delay" milliseconds
+ * @param {int} delay number of milliseconds to delay
+ * @returns {int} the id of the timer so that it can be stopped in the future.
+ */
+function setInterval(func, delay) {
+    var cb_func;
+    var bind_args;
+    var timer_id;
+    var evloop = EventLoop;
+
+    if (typeof delay !== 'number') {
+        throw new TypeError('delay is not a number');
+    }
+    delay = Math.max(evloop.minimumDelay, delay);
+
+    if (typeof func === 'string') {
+        // Legacy case: callback is a string.
+        cb_func = eval.bind(this, func);
+    } else if (typeof func !== 'function') {
+        throw new TypeError('callback is not a function/string');
+    } else if (arguments.length > 2) {
+        // Special case: callback arguments are provided.
+        bind_args = Array.prototype.slice.call(arguments, 2); // [ arg1, arg2, ... ]
+        bind_args.unshift(this); // [ global(this), arg1, arg2, ... ]
+        cb_func = func.bind.apply(func, bind_args);
+    } else {
+        // Normal case: callback given as a function without arguments.
+        cb_func = func;
+    }
+
+    timer_id = evloop.nextTimerId++;
+
+    evloop.insertTimer({
+        id: timer_id,
+        oneshot: false,
+        cb: cb_func,
+        delay: delay,
+        target: Date.now() + delay
+    });
+
+    return timer_id;
+}
+
+/**
+ * stop a repeated timer that has been set.
+ * @method
+ * @param {int} timer_id the id of the timer that was created via setInterval
+ */
+function clearInterval(timer_id) {
+    var evloop = EventLoop;
+
+    if (typeof timer_id !== 'number') {
+        throw new TypeError('timer ID is not a number');
+    }
+    evloop.removeTimerById(timer_id);
+}
+
+/* custom call */
+/**
+ * custom call to exit out of the current event loop.
+ */
+function requestEventLoopExit() {
+    EventLoop.requestExit();
+}
+
+// Hook into the update event of the engine and process all the timers
+Atomic.engine.subscribeToEvent('Update', function updateTimer() {
+    EventLoop.processTimers();
+});
+
+// Hook into the postUpdate event of the engine and process all the setImmediate calls
+Atomic.engine.subscribeToEvent('PostUpdate', function updateImmediates() {
+    EventLoop.processImmediates();
```

Comment:
see note about updateTimer, the PostUpdate event also provides a timeStep argument